src/ng/directive/input.js

```diff
@@ -9,8 +9,8 @@
   ngModelMinErr: false,
 */
 
-// Regex code is obtained from SO: https://stackoverflow.com/questions/3143070/javascript-regex-iso-datetime#answer-3143231
-var ISO_DATE_REGEXP = /\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d\.\d+([+-][0-2]\d:[0-5]\d|Z)/;
+// Regex code was initially obtained from SO prior to modification: https://stackoverflow.com/questions/3143070/javascript-regex-iso-datetime#answer-3143231
+var ISO_DATE_REGEXP = /^\d{4,}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d\.\d+([+-][0-2]\d(:?[0-5]\d)?|Z)$/;
```

Comment:
> Likely to be a problem if the input model is populated directly with an ISO formatted text string

The `ISO_DATE_REGEXP` is only used in an `ngModelController` parser, meaning it will only be applied when parsing the `$viewValue` to transform it to a `$modelValue`. The only case where the `$viewValue` can be an ISO-8601 date string, is on browsers that do not natively support `input[date]` and the users enter it by hand (which is a rather unlikely interaction imo).

Anyway, assuming we still need `ISO_DATE_REGEXP`, what needs to be done in order to properly support all valid timezone offsets on all supported browsers, is "fixing up" the date string before passing it to `new Date()`. That would include adding a colon if missing and add `00` minutes if omitted.

Thinking about it again, this is totally unrelated to the initial intent of this PR, so it's not a good idea to mix them. I suggest, leaving only changes to add support for >4-digit years in this PR and if you fancy feel free to submit another one with the required changes to properly support all valid timezone offset formats in another PR.

Sorry, I had you make those changes and now telling they don't belong here :disappointed: